eslint/local-eslint-config/profile/_common.js

```diff
@@ -101,6 +103,42 @@ function buildRules(profile) {
             ]
           ],
 
+          // Docs: https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/naming-convention.md
+          '@typescript-eslint/naming-convention': [
+            'warn',
+            ...macros.expandNamingConventionSelectors([
+              {
+                selectors: ['method'],
+                modifiers: ['async'],
+                enforceLeadingUnderscoreWhenPrivate: true,
+
+                format: null,
+                custom: {
+                  regex: '^_?[a-zA-Z]\\w*Async$',
+                  match: true
+                },
+                leadingUnderscore: 'allow',
+
+                filter: {
+                  regex: [
+                    // Silently accept names with a double-underscore prefix; we would like to be more strict about this,
+                    // pending a fix for https://github.com/typescript-eslint/typescript-eslint/issues/2240
+                    '^__',
+                    // This is a special exception for naming patterns that use an underscore to separate two camel-cased
+                    // parts.  Example:  "checkBox1_onChangedAsync" or "_checkBox1_onChangedAsync"
+                    '^_?[a-z][a-z0-9]*([A-Z][a-z]?[a-z0-9]*)*_[a-z][a-z0-9]*([A-Z][a-z]?[a-z0-9]*)*$',
+                    // Specifically allow ts-command-line's "execute" and "onExecute" function.
+                    '^onExecute$',
+                    '^execute$'
```

Comment:
Should we specifically exclude these?

Reply:
We'll plan to rename `execute` in another release.